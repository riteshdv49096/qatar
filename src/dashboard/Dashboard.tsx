import React from 'react';
import { FlatList, Text, View, Image, StyleSheet, ActivityIndicator } from 'react-native';
import { useSelector } from 'react-redux';
import imageOne from '../assets/image_one.jpeg';

const Dashboard = () => {
    const blogList = useSelector((state) => state.data.data);

    const renderItem = (item: any) => {
        return (
            <View style={s.itemView}>
                <Image source={imageOne} style={s.image} />
                <View style={s.buttonView}>
                    <Text style={{ fontWeight: '700' }}>{item.item.title}</Text>
                    <Text style={{ fontWeight: '200' }}>{item.item.body}</Text>
                </View>
            </View>
        );
    };
    return (
        <View style={s.container}>
            {blogList?.length === 0 ? (
                <View style={s.loader}>
                    <ActivityIndicator size="small" color="#0000ff" />
                </View>
            ) : (
                <FlatList testID="blogList" keyExtractor={(item, index) => item.id + index.toString()} data={blogList} renderItem={(item) => renderItem(item)} bounces={true} showsVerticalScrollIndicator={false} />
            )}
        </View>
    );
};

export default Dashboard;

export const s = StyleSheet.create({
    container: {
        flex: 1,
        padding: 5,
    },
    itemView: {
        flex: 1,
        flexDirection: 'row',
        padding: 10,
        borderRadius: 5,
        borderWidth: 1,
        borderColor: 'white',
        margin: 5,
        backgroundColor: 'white',
    },
    image: {
        height: 100,
        width: 100,
        resizeMode: 'contain',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 10,
    },
    buttonView: {
        flex: 1,
        paddingHorizontal: 10,
    },
    loader: {
        justifyContent: 'center',
        alignItems: 'center',
        flex: 1,
    },
});
