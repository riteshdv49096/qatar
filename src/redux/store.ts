import { applyMiddleware, legacy_createStore } from 'redux';
import rootReducer from './reducers/reducer';
import thunk from 'redux-thunk';

const configureStore = legacy_createStore(rootReducer, applyMiddleware(thunk));

export default configureStore;
