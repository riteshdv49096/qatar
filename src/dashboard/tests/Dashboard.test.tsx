import { Provider } from 'react-redux';
import Dashboard from '../Dashboard';
import { create } from 'react-test-renderer';
import { Store, AnyAction } from 'redux';
import configureStore from 'redux-mock-store';

describe('Dashboard', () => {
    const initialState = { data: [], isLoading: false, error: null };
    const mockStore = configureStore();
    let store: Store<unknown, AnyAction>;
    test('should show the data in flat list', () => {
        store = mockStore(initialState);
        const tree = create(
            <Provider store={store}>
                <Dashboard navigation={undefined} />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
