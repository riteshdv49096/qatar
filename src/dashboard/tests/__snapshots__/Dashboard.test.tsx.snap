// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should show the data in flat list 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 5,
    }
  }
>
  <RCTScrollView
    bounces={true}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    testID="blogList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
